### ISSUE TITLE
Adding a slimmed down mint function

### ISSUE THREAD

**Issue by Austinhs** [chiru-labs/ERC721A#233]
```solidity
    /**
    * Slimmed down version of _mint for airdrop gas saving
    */
    function _customMint(address to, uint256 quantity) internal {
        unchecked {
            _addressData[to].balance += uint64(quantity);
            _ownerships[_currentIndex].addr = to;

            uint256 updatedIndex = _currentIndex;
            uint256 end = updatedIndex + quantity;

            do {
                emit Transfer(address(0), to, updatedIndex++);
            } while (updatedIndex != end);

            _currentIndex = updatedIndex;
        }
    }
```

I had to copy the ERC721A contract and add this function so I could access the local variables, I used this for airdropping
```solidity
	function airdrop(address[] memory wallets, uint256[] memory amount, uint256 total) external onlyOwner {
		require(totalSupply() + total <= MAX_SUPPLY, "Airdrop will surpass the max supply");
		uint256 walletCount = wallets.length;

		for(uint256 i; i < walletCount; i++) {
			_customMint(wallets[i], amount[i]);
		}
	}
```

This slimmed down the gas on a total airdrop of 5,050 by a few million gas, for anyone wanting to airdrop with ERC721A what do you think is the best way to add this? Should we just add it directly to ERC721A as a basically `unsafeMint`, or any other ideas?

Or am I just being stupid 😅

**Comment by Austinhs** [comment]
Also for what its worth: I fit 5,050 airdrops into 5 transactions by chunking the results, this could do a lot more tokens per transaction if there we were more batching but I'd say we we're nearly 50/50 with multiple and single mints in the airdrop.

Here is a TX for one of the airdrops if you wanted to look into it:
https://etherscan.io/tx/0x85965dfb6fc27153276505b18f3efea96883c949f482c7f9ea8243110dfdf418

**Comment by cygaar** [comment]
While I appreciate the experimentation, I don't think this should be added into ERC721A. Our goal is to provide robust interfaces for users to utilize - this function just removes a couple calls from the existing _mint function so I don't see it adding a ton of value. We don't want to add a ton of experimental features into ERC721A unless there's going to be significant value-add

**Comment by Austinhs** [comment]
> While I appreciate the experimentation, I don't think this should be added into ERC721A. Our goal is to provide robust interfaces for users to utilize - this function just removes a couple calls from the existing _mint function so I don't see it adding a ton of value. We don't want to add a ton of experimental features into ERC721A unless there's going to be significant value-add

Fair, I suppose we are trying to follow OZ as close as possible and you don't see ERC721 with a function like this to cut out things that won't be needed. 

I would say its significant value because of the amount of gas it saves is a ton when doing something at scale. I figured it would be nice to add it as a feature so people don't have to re-invent the wheel and parse the `_mint` function.

Because this does not align with following how OZ developed ERC721 though, I will close this issue

**Comment by cygaar** [comment]
Let me rephrase - I do see the value add here, but I don't want to confuse users on what mint functions they should be using. There are probably other calls that we can remove to further reduce gas in our contract, but we want to find the balance between usability and gas optimizations.

**Comment by Vectorized** [comment]
@cygaar I actually wanna add `_mint(to, quantity)`, because everyone simply uses `_mint(to, quantity, "", false)`, which is ugly imo.

**Comment by Austinhs** [comment]
Going to reopen this issue, as @Vectorized has a pending opinion that has not been challenged. Would like to see @cygaar's opinion before closing the issue again.

**Comment by cygaar** [comment]
We can keep this issue open and wait to see what @Vectorized has in mind
